modules/library/main/src/main/java/org/geotools/data/store/ContentFeatureStore.java

```diff
@@ -283,7 +283,7 @@ FeatureId addFeature(
         // pass through the fid if the user asked so
         boolean useExisting =
                 Boolean.TRUE.equals(feature.getUserData().get(Hints.USE_PROVIDED_FID));
-        if (getQueryCapabilities().isUseProvidedFIDSupported() && useExisting) {
```

Comment:
This change needs to be removed as well. You need to implement a custom QueryCapabilties in the WFSFeatureSource, but instead of having it return "true", if you need to check if the remove WFS server can support the idgen attribute:

https://github.com/geotools/geotools/blob/dad50dbda0d4204ce5e5dae099c3c1da3c096e4a/modules/library/main/src/main/java/org/geotools/data/memory/MemoryFeatureStore.java#L103

Reply:
we must look on section Transaction in getCapabilities and if we have value UseExisting in parameter idgen and after this  set in our realisation QueryCapabilties useProvidedFIDSupported = true, right?